lib/notifier.rb

```diff
@@ -32,7 +32,22 @@ def run
     end
 
     payloads.each do |message_payload|
-      HTTP.post(@slack_url, body: JSON.dump(message_payload))
+      if ENV.has_key? "SLACK_TOKEN"
+        response = HTTP
+          .auth("Bearer #{ENV['SLACK_TOKEN']}")
+          .post("https://slack.com/api/chat.postMessage", json: message_payload)
+          .parse
+
+        if !response["ok"] then
+          if response["error"] == "invalid_auth" then
+            raise "Unable to post to Slack: SLACK_TOKEN is not valid"
+          else
+            puts "Unable to post to Slack: #{response['error']}"
+          end
+        end
```

Comment:
Is it worth extracting this into a helper method to make it a bit more readable?

Reply:
I definitely considered it... but I found in practice it was less readable for me, and had no other benefit. I think the reason is that this class is doing two different things, and in my opinion the `run` method should really be moved out into its own module; so moving the code to send Slack messages was kind of a backward step for me.

I would like to spend some time refactoring the code in this repo along these lines, but I don't know whether it would be a good use of my time or of the time of any reviewers, so I dropped the idea and tried to keep this PR as minimal as my conscience would let me 😅